pkg/collector/pod_collector.go

```diff
@@ -152,3 +170,22 @@ func getPodLabelSelector(client kubernetes.Interface, hpa *autoscalingv2.Horizon
 
 	return nil, fmt.Errorf("unable to get pod label selector for scale target ref '%s'", hpa.Spec.ScaleTargetRef.Kind)
 }
+
+// GetPodReadyAge extracts corev1.PodReady condition from the given pod object and
+// returns true, time.Duration() for pod.LastTransitionTime if the condition corev1.PodReady is found. Returns time.Duration(0s), false if the condition is not present.
+func GetPodReadyAge(pod corev1.Pod) (bool, time.Duration) {
+	t := time.Now()
+	podReadyAge := time.Duration(0 * time.Second)
+	conditions := pod.Status.Conditions
+	if conditions == nil {
+		return false, podReadyAge
+	}
+	for i := range conditions {
+		if conditions[i].Type == corev1.PodReady {
+			podReadyAge = time.Duration(t.Sub(conditions[i].LastTransitionTime.Time).Nanoseconds())
```

Comment:
nit: you can simplify this to `time.Since`: https://golang.org/pkg/time/#Since

Reply:
Used time.Since